src/texture_wrapper.rs

```diff
@@ -31,16 +35,22 @@ impl TextureWrapperManager {
         img: &DynamicImage,
         gfx: &mut Graphics,
         settings: &PersistentSettings,
-    ) {
+    ) -> Result<(), String> {
         //First: try to update an existing texture
         if let Some(tex) = &mut self.current_texture {
             if tex.width() as u32 == img.width()
                 && tex.height() as u32 == img.height()
                 && img.color() == tex.image_format
             {
                 debug!("Re-using texture as it is the same size and type.");
-                tex.update_textures(gfx, img);
-                return;
+                match tex.update_textures(gfx, img) {
```

Comment:
Not required, just a hint:

you can just use the Err if you ignore the Ok() part in match:

```rust
if let Err(error) = tex.update_textures(gfx, img) {
    self.clear();
    error!("{error}");
    return Err(error);
}
```